src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -0,0 +1,607 @@
+        groupby["project"] = Column("project_id")
+
+    data: List[Tuple[_MetricName, _SnubaData]] = []
+
+    #: Find the field that needs a specific column in a specific metric
+    metric_to_output_field: MutableMapping[
+        Tuple[_MetricName, _VirtualColumnName], _OutputField
+    ] = {}
+
+    def get_query(
+        entity_key: EntityKey,
+        metric_id: int,
+        columns: Sequence[str],
+        series: bool,
+        extra_conditions: List[Condition],
+        remove_groupby: Set[Column],
+    ) -> Query:
+        """Build the snuba query"""
+        full_groupby = set(groupby.values()) - remove_groupby
+        if series:
+            full_groupby.add(Column(TS_COL_GROUP))
+
+        where = conditions + [Condition(Column("metric_id"), Op.EQ, metric_id)] + extra_conditions
+
+        return Query(
+            dataset=Dataset.Metrics.value,
+            match=Entity(entity_key.value),
+            select=[Column(column) for column in columns],
+            groupby=list(full_groupby),
+            where=where,
+            granularity=Granularity(query.rollup),
+        )
+
+    # Referrers must be searchable, so no string interpolation here
+    referrers = {
+        "user": {
+            "series": "release_health.metrics.sessions_v2.user.series",
+            "totals": "release_health.metrics.sessions_v2.user.totals",
+        },
+        "session.duration": {
+            "series": "release_health.metrics.sessions_v2.session.duration.series",
+            "totals": "release_health.metrics.sessions_v2.session.duration.totals",
+        },
+        "session": {
+            "series": "release_health.metrics.sessions_v2.session.series",
+            "totals": "release_health.metrics.sessions_v2.session.totals",
+        },
+        "session.error": {
+            "series": "release_health.metrics.sessions_v2.session.error.series",
+            "totals": "release_health.metrics.sessions_v2.session.error.totals",
+        },
+    }
+
+    def get_query_data(
+        entity_key: EntityKey,
+        metric_name: _MetricName,
+        metric_id: int,
+        columns: Sequence[str],
+        extra_conditions: Optional[List[Condition]] = None,
+        remove_groupby: Optional[Set[Column]] = None,
+    ) -> Generator[Tuple[_MetricName, _SnubaData], None, None]:
+        if extra_conditions is None:
+            extra_conditions = []
+
+        if remove_groupby is None:
+            remove_groupby = set()
+
+        for query_type in ("series", "totals"):
+            snuba_query = get_query(
+                entity_key,
+                metric_id,
+                columns,
+                series=query_type == "series",
+                extra_conditions=extra_conditions,
+                remove_groupby=remove_groupby,
+            )
+            referrer = referrers[metric_name][query_type]
+            query_data = raw_snql_query(snuba_query, referrer=referrer)["data"]
+
+            yield (metric_name, query_data)
```

Comment:
What is the point of using a generator here?

Reply:
What's the argument against using a generator here? I.e., can you reformulate as "I would not use a generator here, because..."?